### PR TITLE
Open one dropdown menu at a time 

### LINKED ISSUE #688: dropdown bug
When a dropdown menu is opened, and other is clicked, the first dropdown that was opened should be closed. 
2 dropdown menus should not be opened at the same time.

image

### PR DESCRIPTION
**Pull request by devchoplife**
FIxes #688 
This PR ensures that only one dropdown menu is open at a time on the following pages:
- Transactions page 
- Ticket list page

### CODE REVIEW

File: ui/decredmaterial/dropdown.go [hunk 1]
```diff
@@ -86,6 +86,18 @@ func (d *DropDown) Len() int {
...
+	if dropdowns[0].isOpen {
+		dropdowns[1].isOpen, dropdowns[2].isOpen = false, false
+	} else if dropdowns[1].isOpen {
+		dropdowns[0].isOpen, dropdowns[2].isOpen = false, false
+	} else if dropdowns[2].isOpen {
+		dropdowns[0].isOpen, dropdowns[1].isOpen = false, false
+	}
+}
+*/
+
```
Comment (dreacot) [inline ui/decredmaterial/dropdown.go]: remove comment

File: ui/decredmaterial/dropdown.go [hunk 2]
```diff
@@ -263,3 +275,22 @@ func (d *DropDown) drawLayout(gtx C, body layout.Widget) D {
 
 	return d.linearLayout.Layout2(gtx, body)
 }
+
+//Reslice the dropdowns
```
Comment (dreacot) [inline ui/decredmaterial/dropdown.go]:
```suggestion
// Reslice the dropdowns
```

File: ui/decredmaterial/dropdown.go [hunk 3]
```diff
@@ -263,3 +275,22 @@ func (d *DropDown) drawLayout(gtx C, body layout.Widget) D {
...
+func ResliceDropdown(dropdowns []*DropDown, indexToRemove int) []*DropDown {
+	dropdowns[indexToRemove] = dropdowns[len(dropdowns)-1]
+	return dropdowns[:len(dropdowns)-1]
+}
+
+//Display one dropdown at a time
```
Comment (dreacot) [inline ui/decredmaterial/dropdown.go]:
```suggestion
// Display one dropdown at a time
```

**Review by dreacot — CHANGES_REQUESTED** [review]
Also noticed when the status filter dropdown is opened, you can't click other dropdowns until it's closed.
Other dropdowns should be clickable without having to close the status filter first

Screenshot from 2021-11-17 15-48-29

**Comment by devchoplife** [comment]
> Also noticed when the status filter dropdown is opened, you can't click other dropdowns until it's closed. Other dropdowns should be clickable without having to close the status filter first
> 
> Screenshot from 2021-11-17 15-48-29

I can reproduce this on master too. I will try to find the source of the issue. However, I think the backdrop might be responsible

**Comment by dreacot** [comment]
Yeah, it's on master, i was just stating since you were working on the dropdowns, if it's something complex to fix then we can merge this PR and create and issue for it, just let me know

**Comment by devchoplife** [comment]
> Yeah, it's on master, i was just stating since you were working on the dropdowns, if it's something complex to fix then we can merge this PR and create and issue for it, just let me know

Duly noted

**Review by dreacot — APPROVED** [review]
LGTM

**Pull request merged**